app/models/manageiq/providers/physical_infra_manager.rb

```diff
@@ -34,6 +34,7 @@ class << model_name
     def self.ems_type
       @ems_type ||= "physical_infra_manager".freeze
     end
+    delegate :ems_type, :to => :class
```

Comment:
I don't think you need this if your virtual_delegate uses `emstype` instead of `ems_type`.

With a fake provider added to the ext_manage_systems table you see:
```yaml
irb(main):037:0> p
=> #<ManageIQ::Providers::Redfish::PhysicalInfraManager id: 50, name: "RedFish", created_on: "2019-04-26 22:08:06", updated_on: "2019-04-26 22:08:06", guid: "ffec646a-3145-4e63-8bbf-546c4525b344", zone_id: nil, type: "ManageIQ::Providers::Redfish::PhysicalInfraManager", api_version: nil, uid_ems: nil, host_default_vnc_port_start: nil, host_default_vnc_port_end: nil, provider_region: nil, last_refresh_error: nil, last_refresh_date: nil, provider_id: nil, realm: nil, tenant_id: 1, project: nil, parent_ems_id: nil, subscription: nil, last_metrics_error: nil, last_metrics_update_date: nil, last_metrics_success_date: nil, tenant_mapping_enabled: nil, enabled: true, options: nil, zone_before_pause_id: nil, last_inventory_date: nil>

irb(main):038:0> p.emstype
=> "redfish_ph_infra"

irb(main):039:0> p.class.ems_type
=> "redfish_ph_infra"
```

If this works you will also need to adjust the automate engine and content PRs.

Reply:
Nice, works like a charm, updated, thanks 👍